### PR TITLE
Add code check gomega.Expect(bool).To(gomega.BeTrue(),explain)

### PR DESCRIPTION
**Pull request by tanjunchen**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

This pr adds code check to check 

```
gomega.Expect(bool).To(gomega.BeTrue(),explain)  
gomega.Expect(bool).To(gomega.BeFalse(),explain)  
```
 in e2e tests.

Remid that we should use framework.Equal(bool,true(or false),explain...)

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

**Comment by tanjunchen** [comment]
/hold 
due to  https://github.com/kubernetes/kubernetes/pull/86147  https://github.com/kubernetes/kubernetes/pull/86156  https://github.com/kubernetes/kubernetes/pull/86148

/cc @oomichi 
/cc @liggitt 
/cc @neolit123 
Thanks

File: hack/verify-test-code.sh [hunk 1]
```diff
@@ -51,6 +51,30 @@ do
     fi
 done
 
+errors_expect_be_bool=()
+for file in "${all_e2e_files[@]}"
+do
+    if grep -E "gomega\.Expect\(.*\)\.(To|NotTo)\(gomega\.(BeTrue|BeFalse)\(\)" "${file}" > /dev/null
```
Comment (neolit123) [inline hack/verify-test-code.sh]: do we need to have the `gomega` as part of the call line?
see here it's missing https://github.com/kubernetes/kubernetes/pull/86157/files#diff-40c7fe92055b0c9cbceaf4e26af9a740R39

File: hack/verify-test-code.sh [hunk 2]
```diff
@@ -51,6 +51,30 @@ do
...
+if [ ${#errors_expect_be_bool[@]} -ne 0 ]; then
+  {
+    echo "Errors:"
+    for err in "${errors_expect_be_bool[@]}"; do
+      echo "$err"
+    done
+    echo
+    echo 'The above files need to use framework.ExpectEqual(foo, true, explain) or '
+    echo 'framework.ExpectEqual(foo, false, explain) instead of '
+    echo 'gomega.Expect(foo).To(gomega.BeTrue(), explain) or gomega.Expect(foo).To(gomega.BeFalse(), explain)'
```
Comment (neolit123) [inline hack/verify-test-code.sh]: can the message include examples like so?

`framework.ExpectEqual(foo, true|false, explain)`
`gomega.Expect(foo).To(gomega.BeTrue()|gomega.BeFalse(), explain`

**Comment by neolit123** [comment]
@tanjunchen did someone request this change?
is there a tracking issue?

**Comment by neolit123** [comment]
@oomichi do we think we should force usage of our bool wrapper in all locations?

**Comment by pohly** [comment]
> do we think we should force usage of our bool wrapper in all locations?

What is the argument in favor of that? There's no "why we want this" in the PR or commit description.

The  way I see it, the `framework` variant is shorter, but it's not "better". In a test that that already has several "normal" Gomega assertions it looks odd to use the framework for those checks that have wrappers.

IMHO we shouldn't force test developers to use the framework version.

**Comment by neolit123** [comment]
> IMHO we shouldn't force test developers to use the framework version.

i agree.

> What is the argument in favor of that? There's no "why we want this" in the PR or commit description.

@tanjunchen do you have a proposal of why using the wrapper is better everywhere?

**Comment by liggitt** [comment]
I agree unilateral restrictions on test code like this need agreement/justification

**Comment by tanjunchen** [comment]
First of all, I apologize for having proposed this pr without any proposal or discussion.

Secondly, open this pr, I think there are ```Expect\(.*\)\.To\((gomega\.Equal|Equal)```   ```Expect\(.*\)\.(NotTo|ToNot)\(.*HaveOccurred\(\)```    and other restrictions  in k/k  test/ , like https://github.com/kubernetes/kubernetes/blob/master/hack/verify-test-code.sh#L30 why not use existing framework.ExpectEqual to handle Boolean values.

Thirdly, it is not good to restrict developers too much. I just think it may be better to use existing wrappers.

In the end, this is just my personal idea, and no one has suggested that I can do it like this.
Thanks,for everyBody.

/close  for now

**Pull request closed without merging**